modules/core/core-image/src/main/java/com/enonic/xp/core/impl/image/ImageServiceImpl.java

```diff
@@ -53,67 +51,47 @@
     public ByteSource readImage( final ReadImageParams readImageParams )
         throws IOException
     {
-        final Path cachedImagePath = getCachedImagePath( readImageParams );
-        ByteSource imageByteSource = ImmutableFilesHelper.computeIfAbsent( cachedImagePath, () -> createImage( readImageParams ) );
-        return imageByteSource;
+        final String format = doGetFormatByMimeType( readImageParams.getMimeType() );
```

Comment:
It seems unnecessary to extract format from MIME type here as we only ignore processing very well defined `image/gif` MIME. So we can remove `renderAsASource` altogether and replace it with plain `"image/gif".equals(readImageParams.getMimeType())`